### PR TITLE
feat: android

### PR DESCRIPTION
**Pull request by blu3beri**
Signed-off-by: Berend Sliedrecht <berend@animo.id>

### CODE REVIEW

**Comment by blu3beri** [comment]
@TimoGlastra PR for android :)

No clean up done yet. Might be some minor things we can remove / rename. However this seems to work.

Could you test this at some point? (maybe I have something cached... who knows)

File: lib/cpp-generated/NativeModules.cpp [hunk 1]
```diff
@@ -16,8 +16,7 @@ static jsi::Value __hostFunction_MultiplyModuleCxxSpecJSI_multiply(jsi::Runtime
   return static_cast<MultiplyModuleCxxSpecJSI *>(&turboModule)->multiply(rt, args[0].getNumber(), args[1].getNumber());
 }
 
-MultiplyModuleCxxSpecJSI::MultiplyModuleCxxSpecJSI(std::shared_ptr<CallInvoker> jsInvoker)
-  : TurboModule("MultiplyModule", jsInvoker) {
+MultiplyModuleCxxSpecJSI::MultiplyModuleCxxSpecJSI(std::shared_ptr<CallInvoker> jsInvoker): TurboModule("MultiplyModule", jsInvoker) {
   methodMap_["multiply"] = MethodMetadata {2, __hostFunction_MultiplyModuleCxxSpecJSI_multiply};
 }
```
Comment (TimoGlastra) [inline lib/cpp-generated/NativeModules.cpp:21]: Is this manually added to the generated code?

File: src/index.ts [hunk 2]
```diff
@@ -1,6 +1,12 @@
 import type { Spec } from './turboModule'
 
+// TODO: we need to do some proper initialization
+//       iOS every other reload does not work
+//       Maybe we need to add a timeout?
+//       Or just try it 3 times.
```
Comment (TimoGlastra) [inline src/index.ts:6]: Do you know whether the turbo module is re-initialized on hot reload? With native modules the native side is not reloaded, only the js side. But with shared objects etc I can imagine that we need to reload both sides now right?

**Review by TimoGlastra — APPROVED** [review]
Nice!

File: lib/cpp-generated/NativeModules.cpp (hunk 1 above)
Comment (blu3beri) [reply lib/cpp-generated/NativeModules.cpp:21]: Manually generated. I removed an indent

File: src/index.ts (hunk 2 above)
Comment (blu3beri) [reply src/index.ts:6]: It has something to do with the global not being available. I am not too sure when that happens but it is inconsistent on iOS.

File: lib/cpp-generated/NativeModules.cpp (hunk 1 above)
Comment (TimoGlastra) [reply lib/cpp-generated/NativeModules.cpp:21]: Ah I thought the `TurboModule("MultiplyModule", jsInvoker) ` part was added, but it's on the same line now 👍

**Pull request merged**